src/engine/enginebufferscalelinear.cpp

```diff
@@ -123,20 +120,64 @@ CSAMPLE* EngineBufferScaleLinear::getScaled(unsigned long buf_size) {
         m_dOldRate = 0.0;
         m_dRate = rate_add_new;
         // pass the address of the sample at the halfway point
-        do_scale(&m_buffer[buf_size / 2], buf_size / 2, &samples_read);
+        samples_read += do_scale(&pOutput[buf_size / 2], buf_size / 2);
+        return samples_read;
+    }
+
+    samples_read += do_scale(pOutput, buf_size);
```

Comment:
This invocation of do_scale() should be enclosed in an else {} block to improve readabilty. Then you also need only a single return statement. I'm a fan of early exits, but not in this particular case ;)